src/bitcoin-client.js

```diff
@@ -28,39 +31,53 @@ const sumValueOfUtxos = (utxos) =>
   utxos.reduce((total, utxo) => total + utxo.value, 0);
 
 /**
- * Select UTXOs from the set to cover the value and fee.
+ * Selects UTXOs from the set to cover the value and fee.
  *
- * UTXOs are sorted ascending by value to pick the smaller ones first. This will
- * increase the cost of the initial transactions but will help reduce the
- * chances of creating dust outputs.
+ * UTXOs are selected sorting the list in ascending by value to pick the smaller
+ * ones first, or are just selected at random.
  */
-function selectUtxos(utxos, value, feeLevel) {
-  utxos.sort((a, b) => a.value - b.value);
+const utxoSelectionStrategies = {
+  random: "random",
+  smallerFirst: "smaller-first",
+};
+function selectUtxos(utxos, value, feeLevel, strategy) {
+  let sorted;
+  switch (strategy) {
+    case utxoSelectionStrategies.smallerFirst:
+      sorted = [].concat(utxos.sort((a, b) => a.value - b.value));
```

Comment:
Not strictly needed but... Since the call to `shuffle()` below returns a new array but `.sort()` modifies the array in place I preferred to force `sorted` to always point to a new array for consistency.

I could have done this instead to prevent modifying `utxos` too but didn't think about it when I wrote the code:

```suggestion
      sorted = utxos.concat([]).sort((a, b) => a.value - b.value);
```